### DISCUSSION TITLE
Deploying activity-browser with preloaded ecoinvent database for teaching purposes

### DISCUSSION THREAD

**Discussion by nicolnt** [LCA-ActivityBrowser/activity-browser#1358] [Q&A]
I am searching for a way to deploy activity browser with preloaded ecoinvent database onto classroom computers at our university.

Is there a way to package a version of activity browser where ecoinvent would already be set up for students?

My objectives are:

1. To avoid setting up the database on every computer in the room myself.
2. To avoid letting the students import a copy of the database file by themselves (it would mean that they can copy it).

The IT service can help but I need to give clear instructions on how to proceed.

My intuition would be that there's a way to clone activity-browser data where the database is already imported and deploy this to other computers. I couldn't find where ecoinvent files end up to be able to try myself.

**Comment by mrvisscher** [comment]
Hi @nicolnt,

Projects are saved separate from the Activity Browser install, by default this would be an application folder based on the operating system. In the case of windows: `C:\Users\***\AppData\Local\pylca\Brightway3` I guess you could clone the contents of this folder to all the classroom computers, which in theory should be enough for the AB to recognize the project and it's databases.

Alternatively you could write a script that uses Brightway's restore project functionality to restore a project backup .7z to the application instance in question: https://docs.brightway.dev/en/legacy/content/api/bw2io/backup/index.html 

Kind regards,

Marin

**Reply by nicolnt** [reply]
Thank you Marin for the neat idea!

Here's what I tested from what I understood.
I did a clone of a testing project with ecoinvent loaded `testprojectab.497aa7f3e7552d33a895b8f6518bb0db,` and renamed it `cloneproject` to try if I could see it as a new project.
![image](https://github.com/user-attachments/assets/4bd61fe7-9e8a-4299-bb9f-9d4fe19c4bd4)

But after restarting activity browser, I am unable to see this new clone:
![image](https://github.com/user-attachments/assets/460fa2d8-3e2e-472e-8a00-7d377d25baa6)

**Reply by mrvisscher** [reply]
Hi @nicolnt 

It doesn't show up because it hasn't been registered in the projects.db. Manually cloning only works if you copy the entire `Brightway3` folder and contents. Otherwise use the backup api (which will register the project in the database for you).

**Reply by nicolnt** [reply]
Alright, thank you very much. I'll keep you in touch here on how it went for this deployment for future users in the same situation!

**Reply by nicolnt** [reply]
Hi @mrvisscher 
Everything went OK with this method, so thank you very much!
The only downside is that it's not possible to keep already existing projects, but it's fine for short-term teaching sessions.

**Reply by mrvisscher** [reply]
That's great to hear!

Indeed, if you would want to keep existing projects you would need to do it programmatically, or use another location and change the project directory setting in the Activity Browser.

Kind regards,

Marin
